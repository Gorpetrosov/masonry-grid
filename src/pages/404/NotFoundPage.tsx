import "./NotFoundPage.css"
import { Link } from 'react-router-dom';
import {memo, useEffect} from "react";

interface NotFoundPageProps {
    text?: string;
}

const NotFoundPage = memo(({text = "Uh oh! Looks like you got lost."}: NotFoundPageProps) => {
    useEffect(() => {
        document.body.classList.add('error-page');
        return () => document.body.classList.remove('error-page');
    }, []);
    return (
        <div className="container-404">
            <h1>404 Not Found</h1>
            <p className="zoom-area">{text}</p>
            <p className="zoom-area"> Go back to the homepage if you dare!</p>

            <section className="error-container">
                <span>4</span>
                <span><span className="screen-reader-text">0</span></span>
                <span>4</span>
            </section>
            <div className="link-container">
                <Link to="/"
                   className="custom-link">GO TO MAIN PAGE</Link>
            </div>
        </div>
    )
})

export default NotFoundPage