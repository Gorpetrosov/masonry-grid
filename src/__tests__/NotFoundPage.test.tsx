import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import NotFoundPage from "../pages/404/NotFoundPage.tsx"

describe('NotFoundPage', () => {
    afterEach(() => {
        document.body.className = ''
    })

    it('should match snapshot', () => {
        const { container } = render(
            <MemoryRouter>
                <NotFoundPage />
            </MemoryRouter>
        )
        expect(container).toMatchSnapshot()
    })
})