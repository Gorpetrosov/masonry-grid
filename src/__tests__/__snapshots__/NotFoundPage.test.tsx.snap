// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFoundPage > should match snapshot 1`] = `
<div>
  <div
    class="container-404"
  >
    <h1>
      404 Not Found
    </h1>
    <p
      class="zoom-area"
    >
      Uh oh! Looks like you got lost.
    </p>
    <p
      class="zoom-area"
    >
       Go back to the homepage if you dare!
    </p>
    <section
      class="error-container"
    >
      <span>
        4
      </span>
      <span>
        <span
          class="screen-reader-text"
        >
          0
        </span>
      </span>
      <span>
        4
      </span>
    </section>
    <div
      class="link-container"
    >
      <a
        class="custom-link"
        data-discover="true"
        href="/"
      >
        GO TO MAIN PAGE
      </a>
    </div>
  </div>
</div>
`;
